### PR TITLE
Remove skipping if access_key/secret_key are empty

### PR DESCRIPTION
**Pull request by mjschultz**
Projects should be careful about what they try to upload and use the more
generic `when: repo: owner/name` method instead.

Closes: https://github.com/drone-plugins/drone-s3/issues/11

### CODE REVIEW

File: DOCS.md [hunk 1]
```diff
@@ -22,4 +22,12 @@ publish:
     source: files/to/archive
     target: /target/location
     recursive: true
+    when:
+      repo: drone-plugins/drone-s3
 ```
```
Comment (bradrydzewski) [inline DOCS.md]: I think we can remove this. We have this documented globally for all plugins, and I don't want people to think this is required

Comment (mjschultz) [inline DOCS.md]: The `repo` arg itself doesn't seem to be well documented which is why I included it here (I only see it's existence in the drone/drone/.drone.yml file). However I'd rather have the docs in the main project anyway.

**Comment by tboerger** [comment]
IMHO there must be some validation that it's working with the IAM role **OR** key and secret.

**Comment by mjschultz** [comment]
Do you mean a parameter that explicitly says "use the IAM role"? That is you'd have to provide either (access_key+secret_key) OR (use_iam: true)?

**Comment by mjschultz** [comment]
I've added the logic to verify that the .drone.yml includes either the (access_key and secret_key) or a use_iam flag.

If none are specified it will print a message stating that some must be specified (will not cause a build failure).

If all are specified it treats that as ambiguous, and exits the plugin with a status of 1 (causing a build failure).

**Comment by mjschultz** [comment]
@tboerger @bradrydzewski any feedback/clarifications?

**Comment by nlf** [comment]
lgtm

i think this is a reasonable solution, i do think you should update the docs to make sure it's very clear that `use_iam` cannot be used with the `access_key`/`secret_key` pair though

**Comment by tboerger** [comment]
We should follow https://github.com/drone-plugins/drone-ecr/pull/16#issuecomment-226696026

**Comment by tboerger** [comment]
https://github.com/drone-plugins/drone-s3/pull/19 is also an interesting discussion.

**Comment by josmo** [comment]
with .5 it's moved to using the verified yml for using instance roles. #20 is pulled into the latest. Not sure if we should close this PR unless it's for .4 only?

**Comment by denizs** [comment]
+1

**Comment by tboerger** [comment]
This can be closed, the plugin already uses the default authentication (IAM) if no key and secret is defined: https://github.com/drone-plugins/drone-s3/blob/master/plugin.go#L95-L99

**Pull request closed without merging**